apps/backend/src/modules/groups/group.repository.ts

```diff
@@ -45,13 +49,19 @@ class GroupRepository implements Repository {
 		return Promise.resolve(null);
 	}
 
-	public async findAll(): Promise<{ items: GroupEntity[] }> {
-		const groups = await this.groupModel
+	public async findAll({
+		page,
+		pageSize,
+	}: PaginationQueryParameters): Promise<PaginationResponseDto<GroupEntity>> {
+		const { results, total } = await this.groupModel
 			.query()
+			.page(page, pageSize)
 			.withGraphFetched("[permissions, users]");
 
+		// Return the paginated response with mapped entities
```

Comment:
I think that this comment can be removed, it is clear what we are returning from the signature.
```suggestion
```

Reply:
Yep, it is now removed from the code